File: Frontend/src/utils/resolveOrderStatus.js
const resolveOrderStatus = (code) => {
  switch (code) {
    case 0:
      return 'Pending';
    case 1:
      return 'Partial';
    case 2:
      return 'Approved';
    case 3:
      return 'Rejected';
    case 4:
      return 'Completed';
    case 5:
      return 'Cancel Requested';
    case 6:
      return 'Cancelled';
    default:
      return 'Unknown';
  }
};

export default resolveOrderStatus;

File: mobile/app/src/main/java/com/example/mobile/utils/ResolveOrderStatus.kt
package com.example.mobile.utils

/*
*
* A function to resolve the order status
* from the integer value to a string value
*
* */

fun resolveOrderStatus(status: Int): String {
    return when (status) {
        0 -> "Pending"
        1 -> "Partial"
        2 -> "Approved"
        3 -> "Rejected"
        4 -> "Completed"
        5 -> "Cancel Requested"
        6 -> "Cancelled"
        else -> "Unknown"
    }
}